### PR TITLE
[enh] Global refactoring

### PR DESCRIPTION
**Pull request by maniackcrudelis**
## Problem
- *Need a whole refactoring*

## Solution
- *Package refactored to adjust to latest standards*

## PR Status
Work finished. Package_check, basic tests and upgrade from last version OK.  
Could be reviewed and tested.  

## Validation
~*Medium decision*~
- [ ] ~**Complete test** :~
- [ ] ~**Upgrade previous version** :~
- [X] ~**Code review** : JimboJoe~
- [ ] ~**Code review** :~
- [X] ~**Approval (LGTM)** : JimboJoe~
- [ ] ~**Approval (LGTM)** :~
- [ ] ~**Approval (LGTM)** :~
- [X] ~**CI succeeded** : JimboJoe~
- [ ] ~**CI succeeded** :~
~One public package check log is enough instead of two checks.  
When the PR is mark as ready to merge, you have to wait for 7 days before really merge it.~

Since this PR has been rerouted from master to testing. It passes to a minor decision.
- [x] **Upgrade previous version** : JimboJoe
- [x] **Code review** : JimboJoe
- [x] **Approval (LGTM)** : JimboJoe
- [x] **Approval (LGTM)** : frju365
- [x] **CI succeeded** : JimboJoe
When the PR is mark as ready to merge, you have to wait for 3 days before really merge it.

### CODE REVIEW

File: scripts/install [hunk 1]
```diff
@@ -1,41 +1,93 @@
...
+ynh_add_nginx_config
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
+#=================================================
+
+# Set permissions to app files
+chown -R www-data: $final_path
```
Comment (JimboJoe) [inline scripts/install]: Shouldn't we create a dedicated user...?

Comment (maniackcrudelis) [reply scripts/install]: No, there's no php in this app. So, you can't use a dedicated user.

Comment (JimboJoe) [reply scripts/install]: At least, you actually don't need that, as it's only client side. Forget that 😉

**Comment by JimboJoe** [comment]
Upgrade from previous version OK 👍

**Comment by JimboJoe** [comment]
Will be merged in 3 days 👍

**Pull request merged**